examples/skybox/main.rs

```diff
@@ -0,0 +1,299 @@
+            "examples/skybox/images/negz.jpg",
+        ];
+        let faces = paths
+            .iter()
+            .map(|path| {
+                let f = std::fs::File::open(path).expect("can open file");
+                let f = std::io::BufReader::new(f);
+                image::load(f, image::JPEG).expect("Can load image").to_rgba()
+            })
+            .collect::<Vec<_>>();
+
+        let (image_width, image_height) = faces[0].dimensions();
+        let texture_extent = wgpu::Extent3d {
+            width: image_width,
+            height: image_height,
+            depth: 1,
+        };
+
+        let texture = device.create_texture(&wgpu::TextureDescriptor {
+            size: texture_extent,
+            array_layer_count: 6,
+            mip_level_count: 6, // [WIP] i'm not sure this is right, but if this is 0 it segfaults
+            sample_count: 1,
+            dimension: wgpu::TextureDimension::D2,
+            format: SKYBOX_FORMAT,
+            usage: wgpu::TextureUsage::SAMPLED
+                | wgpu::TextureUsage::COPY_DST
+                | wgpu::TextureUsage::COPY_SRC,
+        });
+
+        {
+            let mut encoder =
+                device.create_command_encoder(&wgpu::CommandEncoderDescriptor { todo: 0 });
+
+            for (i, image) in faces.iter().enumerate() {
+                let (image_width, image_height) = image.dimensions();
+                println!(
+                    "Copying skybox image {} of size {},{} to gpu",
+                    i, image_width, image_height,
+                );
+                let image_buf = device
+                    .create_buffer_mapped(
+                        image.len(),
+                        wgpu::BufferUsage::COPY_SRC | wgpu::BufferUsage::COPY_DST,
+                    )
+                    .fill_from_slice(&image);
+                encoder.copy_buffer_to_texture(
+                    wgpu::BufferCopyView {
+                        buffer: &image_buf,
+                        offset: 0,
+                        row_pitch: 4 * image_width,
+                        image_height,
+                    },
+                    wgpu::TextureCopyView {
+                        texture: &texture,
+                        mip_level: i as u32, // [WIP] this doesn't seem right either?
+                        array_layer: i as u32,
+                        origin: wgpu::Origin3d {
+                            x: 0.0,
+                            y: 0.0,
+                            z: 0.0,
+                        },
+                    },
+                    texture_extent,
+                );
+            }
+        }
+
+        let texture_view = texture.create_view(&wgpu::TextureViewDescriptor {
+            format: SKYBOX_FORMAT,
+            dimension: wgpu::TextureViewDimension::Cube,
+            aspect: wgpu::TextureAspect::default(),
+            base_mip_level: 0,
+            level_count: 6, // [WIP] this doesn't seem right either
```

Comment:
that should be 1

Reply:
Ok I definitely had this as 1 before and I had validation issues on Linux. I’ll set to the suggested mip values, I might now have tried with mip level 0 though when copying, will try